bin/Color.ml

```diff
@@ -0,0 +1,21 @@
+let ofHexString s =
+  if String.length s = 4 || String.length s = 7 then
```

Comment:
Not that String.length is expensive, but you could also do

```ocaml
let ofHexString s = match String.length s with
  | (4 | 7) as len ->
    let short = len = 4 in
    ...
    (match r, g, b with
    | ...
    | ...)
  | _ -> None
```

(Parens needed for a nested `match` expression)